jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -225,19 +260,23 @@ protected void customize(SSLEngine sslEngine, Request request)
     {
         SSLSession sslSession = sslEngine.getSession();
 
-        if (_sniHostCheck)
+        if (_sniHostCheck || _sniRequired)
         {
             String name = request.getServerName();
             X509 x509 = (X509)sslSession.getValue(SniX509ExtendedKeyManager.SNI_X509);
 
-            if (x509 != null && !x509.matches(name))
+            if (LOG.isDebugEnabled())
+                LOG.debug("Host {} matched SNI {}", name, x509);
```

Comment:
I don't see any matching going on at this point? Maybe the debug should say "matchING" not "matchED"?